### ISSUE TITLE
FileSystem DownloadResumable.downloadAsync fails silently or crashes the app on iOS when large file download nears completion

### ISSUE THREAD

**Issue by gyoung-panopto** [expo/expo#8395]
### Environment

Expo CLI 3.17.24 environment info:
    System:
      OS: Windows 10 10.0.18362
    Binaries:
      Node: 10.16.3 - C:\Program Files\nodejs\node.EXE
      npm: 6.7.0 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  3.3.0.0 AI-182.5107.16.33.5199772
    npmPackages:
      react: ^16.9.0 => 16.9.0

This issue only reproduces on iOS (both in Expo client and in standalone app)

### Steps to Reproduce

Use FileSystem.createDownloadResumable to download a large file, reporting progress with each update like in the snack below

### Expected Behavior

Large files complete downloading without crashing the app just like small files do

### Actual Behavior

On iOS, the app either crashes or the download never completes when nearing completion. The files affected are over 1GB. No errors appear.

### Reproducible Demo

Snack: https://snack.expo.io/T5Qw!7Xzn

I tried creating this snack with the basic workflow of my code to demonstrate the issue. I can't get the file to actually download when running the snack, though.

**Comment by byCedric** [comment]
Hi @gyoung-panopto! Thanks for the snack! I tried to run it locally on a simulator, but it completed without crashing. I had to edit 1 line of code:

```
FileSystem.createDownloadResumable(
  "http://distribution...",
  "bbb_sunflower_native_60fps_normal.mp4",
...
```

The download has to be stored in a folder it can access, in my case I had to add `FileSystem.documentDirectory`, so that would make it:

```
FileSystem.createDownloadResumable(
  "http://distribution...",
  FileSystem.documentDirectory + "bbb_sunflower_native_60fps_normal.mp4",
...
```

Could you check if you did this as well, if not if that completes without a crash?

Hope it helps!

**Comment by gyoung-panopto** [comment]
Thanks for the information! I've updated the snack based on your suggestions and also pointed it toward one of the specific files giving me issues in my app:

https://snack.expo.io/qoDchZltT

The download completes when running in the Appetize.io emulator, but when I try it on my real iOS device the app freezes and then crashes when the download hits around 99%. I also noticed the download is much slower on my actual device - maybe the crash is related to a slower internet connection? Or maybe there's something specific to this file?
